dotnet/src/Connectors/Connectors.HuggingFace/Core/HuggingFaceHttpRequestFactory.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Net.Http;
+using Microsoft.SemanticKernel.Http;
+
+namespace Microsoft.SemanticKernel.Connectors.HuggingFace.Core;
+
+internal sealed class HuggingFaceHttpRequestFactory : IHttpRequestFactory
+{
+    public HttpRequestMessage CreatePost(object requestData, Uri endpoint, string? apiKey)
```

Comment:
1. Do we foresee a necessity to swap the HTTP request factories? What would be the scenarios for that?  
2. Why are we externalizing the request creation methods into a separate class instead of having them in the HuggingFaceClient? Do we plan to reuse the HuggingFaceHttpRequestFactory with different classes?